geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RegionManagementController.java

```diff
@@ -38,8 +39,21 @@
       @RequestBody RegionConfig regionConfig) {
```

Comment:
After some thought I think the logic of mapping `ClusterManagementResult`'s `Status` to an appropriate http response code is questionable. On one hand I think the controller having this much logic is an anti pattern. In addition we were hoping that this api endpoint would be idempotent. Part of the definition of idempotent is that any subsequent calls of the same input has the same result and same output. The spec of having 2 different success codes that differentiate a successful initial create vs a subsequent create seems to contradict our desire to be idempotent. If it helps the CAPI api behaves such that subsequent redundant calls that are successful return the exact same result as the first successful call.

Reply:
In addition I think the extra information we're trying to pack in is a hypothesis that it'll be useful. An alternative to a "201" vs "200' response is to have another endpoint "/v2/region/exists" and empower the user to learn more about the system on their own instead of having our single endpoint be more complex. I think it would be better to have each endpoint do one very specific thing very cleanly than to add more complex business logic

Reply:
Thanks for the thoughtful comments. WRT idempotency: In our context, is the command idempotent if the cluster state doesn't change and the persisted configuration doesn't change, even if the text portion of the command response is different? 

In the legacy geode commands we have considered the command idempotent if nothing in the cluster changes even though them textual result from the command differs between the original and repeat executions.

If we need to not make cluster changes AND respond with identical `ClusterManagementResult` java objects and JSON strings in the REST request repsonse bodies, than that will be difficult to achieve because the underlying operations on the cluster members handle initial and repeat calls differently, even though the repeat calls don't make any changes in the cluster state.

Another point to consider is what is a repeated request. In this initial implementation we are on;y considering the region name being the same. We don't look at the region type parameter.

Reply:
In other words I think `ClusterManagementResult` should be static or close to it maybe?

Reply:
>is the command idempotent if the cluster state doesn't change and the persisted configuration doesn't change, even if the text portion of the command response is different?

After reading the article I'm convinced we should be consistent and idempotency should apply to the server state. I don't think identical `ClusterManagementResult` is necessary. We can close this thread =)